src/server/sched_func.c

```diff
@@ -335,5 +335,21 @@ set_sched_default(pbs_sched* psched) {
 		psched->sch_attr[(int) SCHED_ATR_sched_cycle_len].at_flags = ATR_VFLAG_DEFLT
 				| ATR_VFLAG_SET | ATR_VFLAG_MODCACHE;
 	}
+	if ((psched->sch_attr[(int) SCHED_ATR_sched_state].at_flags & ATR_VFLAG_SET)
+			==0) {
+		psched->sch_attr[(int) SCHED_ATR_sched_state].at_val.at_str =  malloc(SC_STATUS_LEN + 1);
+		if ( psched->sch_attr[(int) SCHED_ATR_sched_state].at_val.at_str == '\0' )
+			return;
+		else {
+			psched->sch_attr[(int) SCHED_ATR_sched_state].at_val.at_str[SC_STATUS_LEN] = '\0';
```

Comment:
This needs to go after the strncpy() not before it.  strncpy() will use all the characters in the buffer and won't null terminate.